test/exe/main_common_test.cc

```diff
@@ -312,6 +327,17 @@ TEST_P(AdminRequestTest, AdminRequestAfterRun) {
   EXPECT_EQ(1, lambda_destroy_count);
 }
 
+TEST_P(AdminRequestTest, AdminRequestGetStatsUnderFilter) {
+  addArg("--filter-stats");
+  addArg("^(cluster\\.|tcp\\.).*"); // ^(cluster\.|tcp\.).*
+  startEnvoy();
+  started_.WaitForNotification();
+  EXPECT_THAT(adminRequest("/stats", "GET"),
+              AllOf(HasSubstr("cluster"), Not(HasSubstr("cluster\\.")), Not(HasSubstr("tcp\\."))));
```

Comment:
nit: can we add a test case to ensure that the latter two do show up when no regex is passed in?

Reply:
Good nit, I've made this explicit in two tests: `AdminRequestTest.AdminRequestGetStatsWithoutFilter` and `AdminRequestTest.adminRequestGetStatsWithFilter`. And good catch, `tcp.*` wasn't appearing in these stats. I've updated it to filter for `server.*` instead.